ndcube/utils/wcs.py

```diff
@@ -121,100 +123,195 @@ def _augment(cls, header, naxis):
+
+    # Determine which axes will be made missing by this slicing
+    # and extract the real world coordinate for each.
+    # For this, use WCS-order variables, i.e. missing_axes and item_wcs_order.
+    dropped_coords = {}
+    for i, slice_element in enumerate(item_wcs_order):
+        # If axis is not missing and the difference between its start and stop params is 1,
+        # then the slicing will cause the axis to be dropped, i.e. become missing.
+        if new_missing_axes[i] is False:
+            if isinstance(slice_element, slice):
+                # Determine the start index.
+                if slice_element.start is None:
+                    slice_start = 0
+                else:
+                    slice_start = slice_element.start
+                # Determine the stop index.
+                if slice_element.stop is None:
+                    # wcs._pixel_shape is a list of the length of each axis.
+                    slice_stop = wcs.pixel_shape[i]
+                else:
+                    slice_stop = slice_element.stop
+            elif isinstance(slice_element, int):
+                slice_start = slice_element
+                slice_stop = slice_element + 1
+            # Determine the slice's step.
+            # (We will use this is a later version of this code to be more thorough.
+            # For now we'll calculate it and not use it.)
+            # if slice_element.step is None:
+                # slice_step = 1
+            # else:
+                # slice_step = slice_element.step
+            slice_step = 1
+            real_world_coords = []
+            # If slice results in the axis being of length 1, is will be dropped.
+            # Calculate its real world coordinate.c
+            if slice_stop - slice_start <= slice_step:
+                # Set up a list of pixel coords as input to all_pix2world.
+                pix_coords = [0] * len(item_wcs_order)
+                # Enter pixel coordinate for this axis.
+                pix_coords[i] = slice_start
+                # Get real world coordinates of i-th axis.
+                real_world_coords = wcs.all_pix2world(*pix_coords, 0)[i]
+                # Get IVOA axis name from CTYPE.
+                axis_name = _get_ivoa_from_ctype(wcs.wcs.ctype[i])
+                # Add dropped coordinate's name, axis and value to dropped_coords dict of dicts.
+                dropped_coords[axis_name] = {"wcs axis": i,
+                                             "value": real_world_coords * wcs.wcs.cunit[i]}
+                new_missing_axes[i] = True
+    # Use item_wcs_order to slice WCS.
+    new_wcs = wcs.slice(item_wcs_order, numpy_order=False)
+    return new_wcs, new_missing_axes, dropped_coords
+
+
+def _get_ivoa_from_ctype(ctype):
+    """
+    Find keys in wcs_ivoa_mapping dict that represent start of CTYPE.
+    Ensure CTYPE is capitalized.
+
+    """
+    keys = list(filter(lambda key: ctype.upper().startswith(key), wcs_ivoa_mapping))
```

Comment:
We need to add a case for when the ```ctype``` if not represented in the ```wcs_ivoa_mapping``` ```dict```.  To be consistent with the behaviour of ```NDCube.world_axis_physical_types```, the ```axis_name``` should be the ```ctype``` if not represented in ```wcs_ivoa_mapping```. The function should therefore be written like this:
```python
keys = list(filter(lambda key: ctype.upper().startswith(key), wcs_ivoa_mapping))
n_keys = len(keys)
if n_keys == 0:
    axis_name = ctype
elif n_keys == 1:
    axis_name = wcs_ivoa_mapping[key[0]]
else:
    raise ValueError("Non-unique CTYPE key.")
return axis_name
```